### PR TITLE
Fix marshalling logic to correctly handle array of aliases

### PR DESCRIPTION
**Pull request by ElizabethMRichardson**
## tldr
We want to handle the situation where the result of a JS expression returns an array. 

## The changes
* Update `MarshalEntries` to check if the result from `expr.EvaluateSingleValue` is empty, rather than checking for an error
  * Recently when moving from CEL to JS, we changed the [behaviour of EvaluateResultType to no longer return an error](https://github.com/incident-io/catalog-importer/commit/e129b5e62ab17cf82a960be2b8d1aff54b6a671a#diff-1aae2de3e6230d839cdd6f350fb7e882af7702b88ed8496632b7f951b457f7a2R199). The logic in the marshaling of arrays of aliases was [relying on this error being returned](https://github.com/incident-io/catalog-importer/blob/17ad757c0c3a159766765da4f8ab9103dc5e05f5/output/marshal.go#L137-L142), and so we were no longer hitting it 
* Update the logic in MarshalEntries that is dealing with arrays to use `expr.EvaluateArray` rather than `expr.EvaluateSingleValue` (an existing bug from before cel -> js we believe)

## Some extra context if needed

We want to be able to handle marshalling aliases in the situation where catalog entries have different numbers of aliases. Currently you can set the alias source like: `aliases: ['$.aliases[0]', '$.aliases[1]','$.aliases[2]']`, but this doesn't work in the situation above where the lengths could differ.
For example, a jsonnet with: 

```
      sources: [
        {
          inline: {
            entries: [
              {
                name: 'Low',
                external_id: 'low',
                description:
                  'Non-essential features that customers are happy to wait for PR to fix issues with.',
                aliases: ['low'],
              },
              {
                name: 'Medium',
                external_id: 'medium',
                description:
                  'Needs immediate triaging but may not require immediate response. This is the default criticality if left unspecified.',
                aliases: ['medium', 'med', 'mid'], 
              },
              {
                name: 'High',
                external_id: 'high',
                description:
                  'Immediate response required as this impacts a critical part of the customer experience.',
                aliases: ['high', 'max'],
              },
            ],
          },
        },
      ],
```
We want to be able to set the source like:

```
source: {
            name: '$.name',
            external_id: '$.external_id',
            aliases: ['$.aliases'],
          },
```
 and have this be handled correctly.

### CODE REVIEW

File: output/marshal.go [hunk 1]
```diff
@@ -130,23 +130,31 @@ func MarshalEntries(ctx context.Context, output *Output, entries []source.Entry,
 		// Try to parse each alias as either a string or a string array, then concat and
 		// dedupe them together.
 		aliases := []string{}
-		for idx, aliasSource := range aliasesSource {
-			toAdd := []string{}
-
-			alias, err := expr.EvaluateSingleValue[string](ctx, aliasSource, entry, logger)
+		if aliasesSource[0] == "$.aliases" && len(aliasesSource) == 1 {
```
Comment (ElizabethMRichardson) [inline output/marshal.go:133]: is there a better way to do this? is this even what we want

Comment (ElizabethMRichardson) [reply output/marshal.go]: (there was a better way, see updated commit)

**Review by pipt — APPROVED** [review]
Looks good to me!
